app/resources/services/integrations/patreon.js

```diff
@@ -124,16 +125,28 @@ exports.connectUser = async (req, res) => {
     user_id: req.profile.id
   }
   try {
+    // TODO: what is coming back from the profile, has the user paid for the subscription,
+    // if user has paid (set db to say users role is tier1)
+    // profile.pledges
+    // save refresh token for patreon auth - allws check for patreon
+    const pledges = req.profile._json.relationships.pledges
+    let role = 'USER'
```

Comment:
we should be able to call the User.addRole function I wrote here: 
https://github.com/streetmix/streetmix/blob/00673e5bce53edb3417681fc3f7a06f9f011860d/app/db/models/user.js#L98

Reply:
I also am worried about updating the roles inside this function. 

Ideally, I think this function should be for linking the accounts. We probably dont want to have to call this function every time we want to check their status, no? So maybe we should eventually have a seperate function like syncAccountStatus or something to update/remove roles as appropriate